daemon/sailjailclient.c

```diff
@@ -0,0 +1,1388 @@
+
+EXIT:
+    g_strfreev(exec_argv);
+    g_clear_error(&err);
+
+    return validated;
+}
+
+static bool
+sailjailclient_test_elf(const char* filename)
+{
+    static const char elf[4] = {0x7f, 'E', 'L', 'F'};
+
+    bool ret = false;
+    FILE *file = fopen(filename, "rb");
+    if( !file ) {
+        log_err("%s: could not open: %m", filename);
+    }
+    else {
+        char data[sizeof elf];
+        if( fread(data, sizeof data, 1, file) != 1 ) {
+            log_err("%s: could not read", filename);
+        }
+        else if( memcmp(data, elf, sizeof data) == 0 ) {
+            ret = true;
+        }
+        fclose(file);
+    }
+    return ret;
+}
+
+static const char sailjailclient_usage_template[] = ""
+"NAME\n"
+"  %s  --  command line utility for launching sandboxed application\n"
+"\n"
+"SYNOPSIS\n"
+"  %s <option> [--] <application_path> [args]\n"
+"\n"
+"DESCRIPTION\n"
+"  This tool gets application lauch permissions from sailjaild and\n"
+"  then starts the application in appropriate firejail sandbox.\n"
+"\n"
+"OPTIONS\n"
+"  -h --help\n"
+"        Writes this help text to stdout\n"
+"  -V --version\n"
+"        Writes tool version to stdout.\n"
+"  -v --verbose\n"
+"        Makes tool more verbose.\n"
+"  -q --quiet\n"
+"        Makes tool less verbose.\n"
+"  -d --desktop=<desktop>\n"
+"        Define application file instead of using heuristics based\n"
+"        on path to launched application\n"
+"\n"
+"SAILJAIL BACKWARDS COMPATIBILITY OPTIONS\n"
+"  -p --profile=<desktop>\n"
+"        Define application file instead of using heuristics based\n"
+"        on path to launched application\n"
+"        (alias for --desktop)\n"
+"  -s, --section=NAME\n"
+"        Sailjail section in the profile [Sailjail|X-Sailjail]\n"
+"        (silently ignored)\n"
+"  -a, --app=APP\n"
+"        Force adding Sailfish application directories\n"
+"        (silently ignored)\n"
+"  -o, --output=OUT\n"
+"        Where to output log (stdout|syslog|glib)\n"
+"        (chooses between syslog and stderr)\n"
+"        (defaults to stderr when executed from shell, syslog otherwise)\n"
+"  -t, --trace=DIR\n"
+"        Enable libtrace and dbus proxy logging\n"
+"        (silently ignored)\n"
+"\n"
+"EXAMPLES\n"
+"  %s -- /usr/bin/bar\n"
+"        Launch application bar using permissions from bar.desktop\n"
+"  %s -d org.foo.bar -- /usr/bin/bar\n"
+"        Launch application bar using permissions from org.foo.bar.desktop\n"
+"\n"
+"COPYRIGHT\n"
+"  Copyright (c) 2021 Open Mobile Platform LLC.\n"
+"\n"
+"SEE ALSO\n"
+"  sailjaild\n"
+"\n";
+
+static const char sailjailclient_usage_hint[] = "(use --help for instructions)\n";
+
+static const struct option long_options[] = {
+    {"help",         no_argument,       NULL, 'h'},
+    {"version",      no_argument,       NULL, 'V'},
+    {"verbose",      no_argument,       NULL, 'v'},
+    {"quiet",        no_argument,       NULL, 'q'},
+    {"desktop",      required_argument, NULL, 'd'},
+    // bw compat
```

Comment:
Did you forget to add profile here?

Reply:
Or, as we discussed elsewhere, we could just avoid this whole --profile. I guess all current apps use the short form?

Reply:
Looks like it. Now "profile" is in there and "desktop" is gone.